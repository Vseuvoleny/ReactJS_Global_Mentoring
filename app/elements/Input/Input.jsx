import React, { forwardRef, memo } from "react";
import PropTypes from "prop-types";
import classess from "./Input.module.scss";

const Input = forwardRef(
  (
    {
      classes,
      title,
      type = "text",
      value = "",
      withLabel = true,
      onChange,
      ...rest
    },
    ref
  ) => {
    const { errors, touched } = rest;

    return (
      <div className={`${classess.input_container} ${classes}`}>
        {withLabel && (
          <label htmlFor={title} className={`${classess.label} ${classes}`}>
            {title}
          </label>
        )}
        <input
          ref={ref}
          type={type}
          id={title}
          value={value}
          className={`${classess.input} ${classes}`}
          onChange={onChange}
          autoComplete="off"
        />
        {errors && touched.title && (
          <div className={`${classess.error - message} ${classes}`}>
            {errors.title}
          </div>
        )}
      </div>
    );
  }
);

Input.propTypes = {
  classes: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.string,
  withLabel: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func,
};

export default memo(Input);
