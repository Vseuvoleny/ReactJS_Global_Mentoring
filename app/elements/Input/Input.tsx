import React, { forwardRef, memo } from "react";
import classess from "./Input.module.scss";

const Input = forwardRef(
  (
    {
      classes,
      title,
      type = "text",
      value = "",
      withLabel = true,
      onChange,
      ...rest
    },
    ref
  ) => {
    const { errors, touched } = rest;

    return (
      <div className={`${classess.input_container} ${classes}`}>
        {withLabel && (
          <label htmlFor={title} className={`${classess.label} ${classes}`}>
            {title}
          </label>
        )}
        <input
          ref={ref}
          type={type}
          id={title}
          value={value}
          className={`${classess.input} ${classes}`}
          onChange={onChange}
          autoComplete="off"
        />
        {errors && touched.title && (
          <div className={`${classess.error - message} ${classes}`}>
            {errors.title}
          </div>
        )}
      </div>
    );
  }
);

export default memo(Input);
