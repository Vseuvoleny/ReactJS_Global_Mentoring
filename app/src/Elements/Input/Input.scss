@import "../../styles/colors/colors.scss";

.input-container {
  & > * {
    display: block;
  }
  .label {
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: $color-corral;
    margin-bottom: 13px;
  }
  .input {
    opacity: 0.8;
    border-radius: 4px;
    font-size: 20px;
    line-height: 24px;
    color: $color-white;
    padding: 18px;
    outline: none;
    border: 0;
    background: rgba(50, 50, 50, 0.948044);
    width: 100%;
    &::-webkit-calendar-picker-indicator {
      color: transparent;
      opacity: 1;
      background-image: url("../../assets/svg/calendar.svg");
      background-size: contain;
    }
  }

  .error-message {
    padding-top: 4px;
    color: $color-corral;
  }
}
