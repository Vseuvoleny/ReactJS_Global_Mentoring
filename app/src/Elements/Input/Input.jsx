import React, { forwardRef, memo } from "react";
import PropTypes from "prop-types";
import "./Input.scss";

const Input = forwardRef(
  (
    {
      classes,
      title,
      type = "text",
      value = "",
      withLabel = true,
      onChange,
      ...rest
    },
    ref
  ) => {
    return (
      <div className={`input-container ${classes}`}>
        {withLabel && (
          <label htmlFor={title} className={`label ${classes}`}>
            {title}
          </label>
        )}
        <input
          ref={ref}
          type={type}
          id={title}
          value={value}
          className={`input ${classes}`}
          onChange={onChange}
          {...rest}
          autoComplete="off"
        />
      </div>
    );
  }
);

Input.propTypes = {
  classes: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.string,
  withLabel: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func,
};

export default memo(Input);
