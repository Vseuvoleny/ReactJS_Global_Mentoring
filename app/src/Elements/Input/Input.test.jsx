import { render, screen } from "@testing-library/react";
import Input from "./Input";
import React from "react";
import userEvent from "@testing-library/user-event";

describe("Input component", () => {
  test("should render Input component", () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });
  test("onChange fn shoul work properly", () => {
    const onChange = jest.fn();
    render(<Input onChange={onChange} />);
    expect(screen.queryByDisplayValue(/value/)).toBeNull();
    userEvent.type(screen.getByTestId("input"), "value");
    expect(onChange).toHaveBeenCalled();
  });
});
