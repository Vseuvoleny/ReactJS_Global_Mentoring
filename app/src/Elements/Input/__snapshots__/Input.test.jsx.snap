// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render Input component 1`] = `
<div>
  <div
    class="input-container undefined"
  >
    <label
      class="label undefined"
    />
    <input
      autocomplete="off"
      class="input undefined"
      data-testid="input"
      type="text"
      value=""
    />
  </div>
</div>
`;
